From 5c5bb3948697f2ca184a03dedd5666eb2de547ba Mon Sep 17 00:00:00 2001
From: Dimitry Andric <dimitry@andric.com>
Date: Sat, 20 Jan 2018 14:34:33 +0000
Subject: [PATCH] Assume the shared library path variable is LD_LIBRARY_PATH on
 systems except Darwin and Windows.  This prevents inserting an environment
 variable with an empty name (which is illegal and leads to a Python
 exception) on any of the BSDs.

git-svn-id: https://llvm.org/svn/llvm-project/clang-tools-extra/trunk@323040 91177308-0d34-0410-b5e6-96231b3b80d8
---
 test/Unit/lit.cfg | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/test/Unit/lit.cfg b/test/Unit/lit.cfg
index fc63afdb..b40e1cae 100644
--- a/test/Unit/lit.cfg
+++ b/test/Unit/lit.cfg
@@ -19,13 +19,12 @@ config.test_exec_root = config.test_source_root
 # ;-separated list of subdirectories).
 config.test_format = lit.formats.GoogleTest('.', 'Tests')
 
-shlibpath_var = ''
-if platform.system() == 'Linux':
-    shlibpath_var = 'LD_LIBRARY_PATH'
-elif platform.system() == 'Darwin':
+if platform.system() == 'Darwin':
     shlibpath_var = 'DYLD_LIBRARY_PATH'
 elif platform.system() == 'Windows':
     shlibpath_var = 'PATH'
+else:
+    shlibpath_var = 'LD_LIBRARY_PATH'
 
 # Point the dynamic loader at dynamic libraries in 'lib'.
 shlibpath = os.path.pathsep.join((config.shlibdir, config.llvm_libs_dir,
-- 
2.17.0.rc2

